src/secontext.c

```diff
@@ -116,18 +119,21 @@ selinux_getfdcon(pid_t pid, int fd, char **secontext, char **expected)
 	if (!proc_pid)
 		return -1;
 
-	char linkpath[sizeof("/proc/%u/fd/%u") + 2 * sizeof(int)*3];
-	xsprintf(linkpath, "/proc/%u/fd/%u", proc_pid, fd);
+	int rc;
+	char buf[PATH_MAX + 1];
 
-	int rc = getfilecon(linkpath, secontext);
+	rc = snprintf(buf, sizeof(buf), "/proc/%u/fd/%u", proc_pid, fd);
+	if ((unsigned int) rc >= sizeof(buf))
+		return -1;
```

Comment:
I think the use of `linkpath` was justified: if that buffer overflows, it's a programming error, not a runtime error.

Reply:
That was just to save some stack space since there are already many buffers in stack.
Let me revert that, even though there is no chance the size of the /proc link > PATH_MAX.